### PR TITLE
Mapmodule user location

### PR DESCRIPTION
**Pull request by okauppinen**
Add location accuracy and error to UserLocationEvent.

Add options to GetUserLocationRequest including addToMap (adds point and accuracy circle to vectorlayer) and PositionOptions for geolocation api (timeout, enableHighAccuracy, maximumAge).

New Start/StopUserLocationTrackingRequests and handlers which uses Geolocation.watchPosition().

Added some functionalities to mapmodule to get geojson objects, track user's location and zoom map by location accuracy.

Changed MyLocationPlugin to use GetUserLocationRequest (with high accuracy option) and listen to UserLocationEvent for errors.

Increased GeolocationPlugins timeout to 30s because Chrome doesn't halt request while asking permission from user.

Note! Start/StopUserLocationTrackingRequests doesn't have API documentation and mylocationplugin doesn't have localization. This PR is made to review code. Missing documentation and localization will be pushed later.

### CODE REVIEW

File: bundles/mapping/mapmodule/MapModuleClass.ol.js [hunk 1]
```diff
@@ -1114,14 +1165,22 @@ export default class MapModule extends AbstractMapModule {
...
         return null;
     }
+    /**
+     * @method getGeoJSONFromFeatures
+     * @param {Array] features
+     * @return {Object} geojson FeatureCollection
+     */
+    getGeoJSONFromFeatures (features) {
+        var olGeoJSON = new olFormatGeoJSON();
+        var geojsonStr = olGeoJSON.writeFeatures(features);
```
Comment (data-ux) [inline bundles/mapping/mapmodule/MapModuleClass.ol.js]: `writeFeaturesObject` returns the GeoJSON as JS object. Using it we would not need to parse the string on the net line.

File: bundles/mapping/mapmodule/AbstractMapModule.js [hunk 2]
```diff
@@ -551,45 +557,109 @@ Oskari.clazz.define(
...
+                    }, navigatorOpts
+                );
+            } else {
+                // browser doesn't support
+            }
+        },
+        stopUserLocationWatch: function () {
+            if (navigator.geolocation) {
+                navigator.geolocation.clearWatch(this._locationWatch);
             }
```
Comment (data-ux) [inline bundles/mapping/mapmodule/AbstractMapModule.js]: Do we need to have the user location tracking code in the MapModule? Could we move it somewehere else?

File: bundles/mapping/mapmodule/MapModuleClass.ol.js [hunk 3]
```diff
@@ -378,7 +380,56 @@ export default class MapModule extends AbstractMapModule {
...
+    getLocationPathCoordinates () {
+        if (this._locationPath) {
+            return this._locationPath.getCoordinates();
+        }
+        return [];
+    }
+    clearLocationPath () {
+        this._locationPath = null;
+    }
     /* <------------- / OL3 specific ----------------------------------- */
```
Comment (data-ux) [inline bundles/mapping/mapmodule/MapModuleClass.ol.js]: Do we need to keep track of the `_locationPath` in the MapModule impl? It's already too big and sprawling.

File: api/mapping/mapmodule/request/StartUserLocationTrackingRequest.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+```javascript
+var sb = Oskari.getSandbox();
+sb.postRequestByName('StartUserLocationTrackingRequest', [{addToMap: 'location', centerMap: 'single'}]);
+```
+
+Start tracking user's location and add path to the map:
+```javascript
+var sb = Oskari.getSandbox();
+sb.postRequestByName('StartUserLocationTrackingRequest', [{addToMap: 'path', centerMap: 'update'}]);
+}
```
Comment (ZakarFin) [inline api/mapping/mapmodule/request/StartUserLocationTrackingRequest.md]: Extra } here ^

File: bundles/mapping/mapmodule/LocationModule.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+                // notify event without position to signal failure
+                var error = errorCodes[errors.code] || errorCodes[2];
+                sandbox.notifyAll(evtBuilder(null, null, null, error));
+                if (typeof errorCB === 'function') {
+                    errorCB(error);
+                }
+            }, navigatorOpts
+        );
+    } else {
+        // browser doesn't support
```
Comment (ZakarFin) [inline bundles/mapping/mapmodule/LocationModule.js]: Send a UserLocationEvent here too with 'unavailable' error code?

File: bundles/mapping/mapmodule/request/StartUserLocationTrackingRequestHandler.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+            featureStyle: featureStyle
+        };
+        Oskari.getSandbox().postRequestByName('MapModulePlugin.AddFeaturesToMapRequest', [features, layerOptions]);
+    },
+    _clearLocationFromMap: function () {
+        Oskari.getSandbox().postRequestByName('MapModulePlugin.RemoveFeaturesFromMapRequest', [null, null, 'USER_LOCATION_LAYER']);
+    },
+    _addPathToMap: function () {
+        const coords = getLocationCoords();
+        console.log(coords);
```
Comment (ZakarFin) [inline bundles/mapping/mapmodule/request/StartUserLocationTrackingRequestHandler.js]: Remove console.log()

**Pull request merged**